internal/observability/middleware.go

```diff
@@ -62,9 +63,49 @@ type Middleware struct {
 	graph graph
 }
 
-// NewMiddleware constructs a Middleware.
-func NewMiddleware(logger *zap.Logger, scope *metrics.Scope, extract ContextExtractor) *Middleware {
-	return &Middleware{newGraph(scope, logger, extract)}
+// Config configures the observability middleware.
+type Config struct {
+	// Logger to which messages will be logged.
+	Logger *zap.Logger
+
+	// Scope to which metrics are emitted.
+	Scope *metrics.Scope
+
+	// Extracts request-scoped information from the context for logging.
+	ContextExtractor ContextExtractor
+
+	// Log level used to log successful inbound and outbound calls.
+	//
+	// Defaults to DebugLevel.
+	SuccessLevel *zapcore.Level
+
+	// Log level used to log failed inbound and outbound calls.
```

Comment:
Mind adding a few examples of errors that fall into this bucket? E.g., low-level network errors.